src/main/java/frc/robot/subsystems/BatterySimulation.java

```diff
@@ -27,39 +20,34 @@ public BatterySimulation(BatterySubsystem batterySubsystem) {
 
   @Override
   public void simulationPeriodic() {
-    //Consistently lowers the voltage
-    RoboRioSim.setVInVoltage(RoboRioSim.getVInVoltage() - voltageToSubtract);
-
-
-
-    //Checks voltage levels. If below a certain amount (minVoltageRedDouble), it crashes the simulation
-    if (RoboRioSim.getVInVoltage() < Constants.minVoltageRedDouble) {
-      System.out.println("Got to assert voltage");
-      assertEquals(true, batterySubsystem.checkRedVoltage());
-    }
-
-    //Checks general timer. If a certain amount of time has passed (timeInSecondsGeneralRed), it crashes the simulation
-    if (batterySubsystem.getGeneralTimer() > Constants.timeInSecondsGeneralRed) {
-      System.out.println("Got to assert general timer");
-      assertEquals(true, batterySubsystem.checkTimer());
-    }
-    //Checks high current timer. If a certain amount of time has passed (timeInSecondsHighCurrentRed), it crashes the simulation
-    if (batterySubsystem.getHighCurrentTimer() > Constants.timeInSecondsHighCurrentRed) {
-      System.out.println("Got to assert high current timer");
-      assertEquals(true, batterySubsystem.checkTimer());
+    double now = Timer.getFPGATimestamp();
+    if (now >= 0 && now < 5.0) {
+      RoboRioSim.setVInVoltage(RoboRioSim.getVInVoltage() - voltageToSubtract);
+      if (RoboRioSim.getVInVoltage() < Constants.minVoltageRedDouble) {
+        System.out.println("Got to assert voltage");
+        assertEquals(false, batterySubsystem.checkRedVoltage());
+      }
+    } else if (now >= 5.0 && now < 25.0) {
+      if (batterySubsystem.getGeneralTimer() > Constants.timeInSecondsGeneralRed) {
+        System.out.println("Got to assert general timer");
+        assertEquals(true, batterySubsystem.checkTimer());
+      }
+    } else if (now >= 25.0 && now < 50.0) {
+      if (batterySubsystem.getHighCurrentTimer() > Constants.timeInSecondsHighCurrentRed) {
+        System.out.println("Got to assert high current timer");
+        assertEquals(true, batterySubsystem.checkTimer());
+      }
     }
-    //Enable the robot after 10 seconds to rest high current timer
-    if (batterySubsystem.getGeneralTimer() > 10.0 ) {
+    // Enable the robot after 10 seconds to test high current timer
+    if (batterySubsystem.getGeneralTimer() > 10.0) {
       DriverStationSim.setEnabled(true);
     }
 
-    //Forcefully sets the voltage when enabling the robot.
+    // Forcefully sets current when robot is enabled to simulate use
     if (DriverStationSim.getAutonomous() == true || DriverStationSim.getEnabled() == true) {
-      batterySubsystem.startHCTimer();
-      //RoboRioSim.setVInVoltage(12.1);
+      RoboRioSim.setVInCurrent(Constants.simCurrentHigh);
     } else {
-      //RoboRioSim.setVInVoltage(14.0);
-      batterySubsystem.stopHCTimer();
+      RoboRioSim.setVInCurrent(Constants.simCurrentLow);
```

Comment:
The constants here seem a little unnecessary since you already have a constant for what is high current. You might do something like RoboRioSim.setVInCurrent(Constants.maxBatteryCurrent+1) for high current and -1 for low.

Reply:
done